File: app/src/main/java/com/moviereel/moviereel/views/introduction/SplashActivity.java
package com.moviereel.moviereel.views.introduction;

import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Typeface;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.TextView;

import com.moviereel.moviereel.R;
import com.moviereel.moviereel.views.MainActivity;

public class SplashActivity extends AppCompatActivity {
    /*FIELDS*/
    private static final String SPLASH_SCREEN_TAG = SplashActivity.class.getSimpleName();
    private TextView appName;
    private ImageView appIcon;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        setContentView(R.layout.splash_activity_layout);

        initializeUICtrls();

        //set the timer
        Thread timer = new Thread(){
            @Override
            public void run(){
                SharedPreferences getPrefs = PreferenceManager.getDefaultSharedPreferences(getBaseContext());
                //create a new boolean and preference and set it to true
                boolean isFirstStart = getPrefs.getBoolean("firstStart", true);
                //if activity has never started before

                if(isFirstStart){
                    //launch this activity
                    Intent intent = new Intent(SplashActivity.this, AppIntroduction.class);
                    startActivity(intent);

                    //make a new shared preferences editor
                    SharedPreferences.Editor editor = getPrefs.edit();

                    //  Edit preference to make it false because we don't want this to run again
                    editor.putBoolean("firstStart", false);

                    //apply the changes
                    editor.apply();
                }

                try{
                    sleep(2000);
                    Intent openMain = new Intent(SplashActivity.this, MainActivity.class);
                    startActivity(openMain);
                }catch(InterruptedException ie){
                    ie.printStackTrace();
                    Log.d(SPLASH_SCREEN_TAG, ie.toString());
                }
            }
        };
        //start timer
        timer.start();
    }

    /**Initializes the User Interface controls
     * obtains the appName, appTag and appIcon variables and finds their ids in the layout*/
    private void initializeUICtrls() {
        appIcon = (ImageView)findViewById(R.id.appicon_splash);
        appName = (TextView)findViewById(R.id.appname_splash);

        //set the fonts
        String fontPath = "fonts/Roboto-Black.ttf";
        Typeface typeface = Typeface.createFromAsset(getAssets(), fontPath);
        appName.setTypeface(typeface);
    }

    /*kill this splash screen to save memory*/
    @Override
    protected void onPause(){
        super.onPause();
        finish();
    }

}/*END*/

File: mobile-ui/src/main/kotlin/com/moviereel/ui/intro/splash/SplashActivity.kt
package com.moviereel.ui.intro.splash

import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import com.moviereel.R
import com.moviereel.presentation.view.splash.SplashPresenter
import com.moviereel.presentation.view.splash.SplashView
import com.moviereel.ui.base.BaseActivity
import com.moviereel.ui.intro.AppIntroduction
import com.moviereel.ui.main.MainActivity
import org.jetbrains.anko.startActivity
import org.koin.android.ext.android.inject
import javax.inject.Inject

class SplashActivity : BaseActivity(), SplashView {

    val splashPresenter: SplashPresenter<SplashView> by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_splash_layout)

        splashPresenter.onAttach(this)
    }

    /**
     * Abstract method that will be implemented by child activity classes
     * used to setup the views in the activity
     */
    override fun setUp() {

    }

    /*kill this splash screen to save memory*/
    override fun onPause() {
        super.onPause()
        finish()
    }

    /**
     * opens the main activity
     */
    override fun openMainActivity() {
        startActivity<MainActivity>()
        finish()
    }

    /**
     * opens the app introduction activity
     * this will be opened if this is the first time this application is being opened
     */
    override fun openAppIntroductionActivity() {
        startActivity<AppIntroduction>()
        finish()
    }

    /**
     * Starts sync service which will be used to update the data that will be used in the application
     */
    override fun startSyncService() {
        // TODO: 01/04/17: start sync service to fetch today's latest movies/series/shows
    }

    override fun onDestroy() {
        splashPresenter.onDetach()
        super.onDestroy()
    }
}
